scalekit/core.py

```
@@ -254,15 +328,24 @@ def grpc_exec(
                     raise ScalekitServerException.promote(exp)
                 try:
                     self.__authenticate_client()
-                    return self.grpc_exec(func, data, retry=retry-1)
+                    return self.grpc_exec(func, data, retry=retry - 1, attempt=attempt + 1, timeout=timeout)
                 except Exception as refresh_exp:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)